gstreamer-sdp/src/s_d_p_media.rs

```diff
@@ -0,0 +1,448 @@
+        }
+    }
+
+    pub fn formats_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_formats_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn get_attribute(&self, idx: u32) -> Option<SDPAttribute> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_attribute(self.to_glib_none().0, idx) )
+        }
+    }
+
+    pub fn get_attribute_val(&self, key: &str) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_attribute_val(self.to_glib_none().0, key.to_glib_none().0))
+        }
+    }
+
+    pub fn get_attribute_val_n(&self, key: &str, nth: u32) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_attribute_val_n(self.to_glib_none().0, key.to_glib_none().0, nth))
+        }
+    }
+
+    pub fn get_bandwidth(&self, idx: u32) -> Option<SDPBandwidth> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_bandwidth(self.to_glib_none().0, idx))
+        }
+    }
+
+    pub fn get_caps_from_media(&self, pt: i32) -> Option<gst::Caps> {
+        unsafe {
+            from_glib_full(ffi::gst_sdp_media_get_caps_from_media(self.to_glib_none().0, pt))
+        }
+    }
+
+    pub fn get_connection(&self, idx: u32) -> Option<SDPConnection> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_connection(self.to_glib_none().0, idx))
+        }
+    }
+
+    pub fn get_format(&self, idx: u32) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_format(self.to_glib_none().0, idx))
+        }
+    }
+
+    pub fn get_information(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_information(self.to_glib_none().0))
+        }
+    }
+
+    pub fn get_key(&self) -> Option<SDPKey> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_key(self.to_glib_none().0))
+        }
+    }
+
+    pub fn get_media(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_media(self.to_glib_none().0))
+        }
+    }
+
+    pub fn get_num_ports(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_get_num_ports(self.to_glib_none().0)
+        }
+    }
+
+    pub fn get_port(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_get_port(self.to_glib_none().0)
+        }
+    }
+
+    pub fn get_proto(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gst_sdp_media_get_proto(self.to_glib_none().0))
+        }
+    }
+
+    pub fn init(&mut self) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_media_init(self.to_glib_none_mut().0))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn insert_attribute(&mut self, idx: i32, attr: &mut SDPAttribute) -> Result<_, ()> {
```

Comment:
This takes ownership of the argument, not just a mutable reference. And you should `mem::forget` it in the end

Reply:
Same for all the other insert/replace, except for the ones taking strings